### PR TITLE
Update java.md with new recommendations about optimizations and profiling.

### PR DESCRIPTION
**Pull request by geoffreyblake**
Add hints to java.md about removing anti-patterns from Java code, such as hot exception code-paths.  Also update the best ways to profile Java code to recommend APerf as the first choice.

### CODE REVIEW

File: java.md [hunk 1]
```diff
@@ -178,11 +178,52 @@ to manage each platform specific Jar.  Release individual architecture specific
 instance to download these released jars and package them into a combined Jar with a final `mvn release:perform`.
 An example of this methd can be found in the [Leveldbjni-native](https://github.com/fusesource/leveldbjni) `pom.xml` files. 
 
+### Remove Anti-patterns
+
+Anti-patterns can impact the performance on any instance family, but the level of impact can be different between instance families.  Below is a list of anti-patterns
+we have observed to impact Graviton positively when removed as much as possible:
```
Comment (wash-amzn) [inline java.md:184]: "we have observed to impact Graviton positively when removed" keeps flipping around positive/negative. I had to read it a second time to make fully parse it.

I recommend something like "Below is a list of anti-patterns we have found to be particularly impactful on Graviton:"

File: java.md [hunk 2]
```diff
@@ -178,11 +178,52 @@ to manage each platform specific Jar.  Release individual architecture specific
...
+
+1.  **Excessive exceptions**: Throwing exceptions and generating stack-traces has been observed to cost up to 2x more on Graviton platforms compared to x86.  We recommend not to use Java exceptions as control flow, and to remove exceptions when they appear in the hot-code path. Identifying hot exceptions can be done using function profilers like Aperf, Async-profiler, or Linux `perf`. Overhead can be mitigated some by using the `-XX:+OmitStackTraceInFastThrow` JVM flag to allow the Java runtime to optimize the exception flow for some hot paths. The best solution is to avoid the exceptions as much as possible.
```
[the remark this answers is not in the document]
Comment (geoffreyblake) [reply java.md]: Fixed.

File: java.md (hunk 1 above)
Comment (geoffreyblake) [reply java.md]: Fixed.

File: java.md [hunk 3]
```diff
@@ -178,11 +178,62 @@ to manage each platform specific Jar.  Release individual architecture specific
...
+tar -zxf aperf-v0.1.13-alpha-aarch64.tar.gz
+
+# Get the latest Async profiler
+wget https://github.com/async-profiler/async-profiler/releases/download/v3.0/async-profiler-3.0-linux-arm64.tar.gz
+tar -zxf async-profiler-3.0-linux-arm64.tar.gz
+cd async-profiler-3.0-linux-arm64
+sudo mkdir -p /opt/bin
+sudo mkdir -p /opt/lib
+sudo cp -a bin/* /opt/bin
+sudo cp -a lib/* /opt/lib
```
Comment (wash-amzn) [inline java.md:218]: It seems unnecesaary to get putting these in system paths if you're still going to require setting the PATH to run aperf. I'd say leave them where they are and set the PATH to the extracted location. It doesn't clutter up random file system paths that way.

It's odd that async-profiler doesn't have a recommended usage, particularly given that it has library dependencies (so requires a correct library path, too).

**Pull request merged**